www/js/control/ProfileSettings.jsx

```diff
@@ -332,14 +594,110 @@ const ProfileSettings = () => {
+                        <Text variant="">{t('general-settings.log-out-warning')}</Text>
+                    </Dialog.Content>
+                    <Dialog.Actions>
+                        <Button onPress={()=>setLogoutVis(false)}>
+                            {t('general-settings.cancel')}
+                        </Button>
+                        <Button onPress={() => {
+                           DynamicConfig.resetConfigAndRefresh();
+                        }}>
+                            {t('general-settings.confirm')}
+                        </Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </Modal>
+
+            {/* dataPending */}
+            <Modal visible={dataPendingVis} onDismiss={()=>setDataPendingVis(false)} transparent={true}>
+                <Dialog visible={dataPendingVis} 
+                        onDismiss={()=>setDataPendingVis(false)} 
+                        style={styles.dialog(colors.elevation.level3)}>
+                    <Dialog.Title>{t('data pending for push')}</Dialog.Title>
+                    <Dialog.Actions>
+                        <Button onPress={()=>{
+                            setDataPendingVis(false);
+                            Logger.log("user refused to re-sync")}}>
+                                {t('general-settings.cancel')}
+                        </Button>
+                        <Button onPress={()=>{
+                            setDataPendingVis(false);
+                            forceSync();}}>
+                                {t('general-settings.confirm')}
+                        </Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </Modal>
+
+            {/* handle no consent */}
+            <Modal visible={noConsentVis} onDismiss={()=>setNoConsentVis(false)} transparent={true}>
+                <Dialog visible={noConsentVis} 
+                        onDismiss={()=>setNoConsentVis(false)} 
+                        style={styles.dialog(colors.elevation.level3)}>
+                    <Dialog.Title>{t('general-settings.consent-not-found')}</Dialog.Title>
+                    <Dialog.Actions>
+                        <Button onPress={()=>{
+                            setNoConsentVis(false);
+                            setNoConsentMessageVis(true)}}>
+                                {t('general-settings.cancel')}
+                        </Button>
+                        <Button onPress={()=>{
+                            setNoConsentVis(false);
+                            // $state.go("root.reconsent"); //don't know how to do this yet
+                            }}>
+                                {t('general-settings.confirm')}
+                        </Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </Modal>
+
+            {/* handle consent */}
+            <Modal visible={consentVis} onDismiss={()=>setConsentVis(false)} transparent={true}>
```

Comment:
Future: I think we can remove this in the future. The consent "number" is currently an IRB protocol number from UC Berkeley's Committee for the Protection of Human Subjects (CPHS). Since I am no longer at Berkeley, this no longer applies to us. And I had some vision initially that the CPHS protocols could change, and we would require people to consent to the new protocol, but that has never yet happened in practice.

We should think about changing terms and conditions midway through a data collection period, but I don't think this is the solution.

We should remove this and redesign it better.